### ISSUE TITLE
[Unity + FMOD] Error when exiting Play Mode while a Steam Audio Source is attached to an FMOD Studio Event Emitter

### ISSUE THREAD

**Issue by Meorge** [ValveSoftware/steam-audio#244]
Video:
https://youtu.be/Rbq3gPS47bI

I created a minimal project using:
- Unity 2021.3.18f1
- FMOD Studio integration 2.02.12 from FMOD's website
- Steam Audio version 4.1.3 from the GitHub releases

If the FMOD Event has the Steam Audio Spatializer attached to it, and the Steam Audio Source component is enabled on the Game Object with the FMOD Studio Event Emitter component, then an error will appear upon exiting Play Mode.

The only modification I made to any of the files was a fix for FMOD, so that it would properly recognize the phonon_fmod plugin. I explained the fix here: https://qa.fmod.com/t/bug-with-loading-bundle-plugins-on-macos-and-the-fix-for-it/19933 
Otherwise, the files are unmodified.

The complete project (both Unity and FMOD Studio) can be downloaded here: https://drive.google.com/file/d/1GO-QiUZTA1doE6APtlb9_RRUS-7CnChp/view?usp=sharing

The error messages themselves, from `Editor.log`, are:
```
[FMOD] assert : assertion: 'mImpl->mRefCount > 0' failed
UnityEngine.StackTraceUtility:ExtractStackTrace () (at /Users/bokken/build/output/unity/unity/Runtime/Export/Scripting/StackTrace.cs:37)
UnityEngine.DebugLogHandler:LogFormat (UnityEngine.LogType,UnityEngine.Object,string,object[])
UnityEngine.Logger:Log (UnityEngine.LogType,object)
UnityEngine.Debug:LogError (object)
FMODUnity.RuntimeUtils:DebugLogError (string) (at Assets/Plugins/FMOD/src/RuntimeUtils.cs:574)
FMODUnity.RuntimeManager:DEBUG_CALLBACK (FMOD.DEBUG_FLAGS,intptr,int,intptr,intptr) (at Assets/Plugins/FMOD/src/RuntimeManager.cs:77)

(Filename: Assets/Plugins/FMOD/src/RuntimeUtils.cs Line: 574)

[FMOD] AsyncManager::asyncThreadLoop : System::update returned error 28.
UnityEngine.StackTraceUtility:ExtractStackTrace () (at /Users/bokken/build/output/unity/unity/Runtime/Export/Scripting/StackTrace.cs:37)
UnityEngine.DebugLogHandler:LogFormat (UnityEngine.LogType,UnityEngine.Object,string,object[])
UnityEngine.Logger:Log (UnityEngine.LogType,object)
UnityEngine.Debug:LogError (object)
FMODUnity.RuntimeUtils:DebugLogError (string) (at Assets/Plugins/FMOD/src/RuntimeUtils.cs:574)
FMODUnity.RuntimeManager:DEBUG_CALLBACK (FMOD.DEBUG_FLAGS,intptr,int,intptr,intptr) (at Assets/Plugins/FMOD/src/RuntimeManager.cs:77)

(Filename: Assets/Plugins/FMOD/src/RuntimeUtils.cs Line: 574)
```

**Comment by someonesomewheredev** [comment]
Hi, I've been having this issue as well. It seems like it might be related to some changes in FMOD's handling of custom data DSP parameters made in 2.02.10. I fixed it by downgrading to 2.02.09.

**Comment by Meorge** [comment]
I tried this (downgrading the FMOD for Unity integration to 2.02.09) and it so far it seems to have worked for the minimal project example! I will try doing the same for my main project and see if the errors go away there.

**Comment by Infinitylt** [comment]
Downgrading bricks my whole unity project. And unity gives missing default layout error.

**Comment by lakulish** [comment]
We have just released v4.4.0, which contains a fix for this issue: https://github.com/ValveSoftware/steam-audio/releases/tag/v4.4.0.
